types/puppeteer/index.d.ts

```diff
@@ -67,9 +67,15 @@ export interface JSONObject {
 }
 export type SerializableOrJSHandle = Serializable | JSHandle;
 
-export type Platform = 'mac' | 'win32' | 'win64' | 'linux';
+/**
+ * We want to maintain intellisense for known values but remain open to unknown values. This type is a workaround for
+ * [Microsoft/TypeScript#29729](https://github.com/Microsoft/TypeScript/issues/29729). It will be removed as soon as
+ * it's not needed anymore.
+ */
+type LiteralUnion<LiteralType> = LiteralType | (string & { _?: never });
```

Comment:
Is this going to break control-flow?  If someone formerly had `if ( firefox ) doA() else doB()`, they could rely on having chrome in the `else` branch.  Now it will be chrome or a tagged string?

Reply:
It won't break control flow and users can continue to write code in the same way as your example. Where it's an input, the JS implementation validates the value.

Reply:
Sorry, I'm not sure I understand.  Here's a concrete [example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAMglsCAnAhgGwKoDs4HssA88iqaAKuBAHxQC8sCy6FkUAPlABQDOwScWAOZQAZFADeUAPoB+AFxQsEAG7IoAXwCUAbgBQEAB5hcSYFFCsAchADuABSS4AJgFcAxmfrEmmHPgIA5G4AFo4AthAB7FABAGZwSBCxuAYBVHqGxqbmlFAA8mhODs7unjEh4ZHRcQlJKQF6uk4QbmgoiVBu+LyKtnYK1vaOrh56za3t0F1YPbiF-fnzw6WNcLFcc0VQAIS09EGhuBEBmhK6UBdQKNzcyMAAYrXJBpybdjpQAPSf+QDWKCBdOpdLo1lwlPYdntyodjqdxOdLtdbqZHolnpwIe9tF8fgBRJCOJBAkGxFxYDx4LBXG53NF1F5gBQ1dH1eHqbRAA) illustrating my concern.  Can you please have a look and let me know what I'm missing?

Reply:
Ah, I get what you mean from your example – you're right that would be a new type error for that case. I'm not too worried about that though, the same would occur if puppeteer added `safari`.

Reply:
Good point, but it's worth noting that that error message would say "chrome | safari is not assignable to chrome" whereas this one will have an unfamiliar branded string construct.